crypto/mem.c

```diff
@@ -329,6 +329,8 @@ size_t CRYPTO_secure_used(void) { return 0; }
 
 void *OPENSSL_secure_malloc(size_t size) { return OPENSSL_malloc(size); }
 
+void *OPENSSL_secure_zalloc(size_t size) { return OPENSSL_zalloc(size); }
```

Comment:
We've established a precedence with `OPENSSL_secure_malloc` of allocating "unsecured"(?) memory, although I think it would be more proper for these to return `NULL`. 

[`CRYPTO_secure_malloc_init` always returns 0](https://github.com/aws/aws-lc/blob/7b383f4e14be4ed792a48585544887aed536f2c4/crypto/mem.c#L324) to indicate our lack of support for secure heaps.

Reply:
Wouldn't that break any codepaths trying to allocate memory when calling `OPENSSL_secure_malloc` or `OPENSSL_secure_zalloc`? It's true that we don't support "secure heaps" today, but I'm not sure if we want to block making integration easier due to the non-support.

Reply:
Returning NULL was my first thought but doing so would break code using these functions and we are creating more instances of de-referencing NULL pointers? We should make a decision on which is more important. However, given that for other functions we "fail" (by returning 0), I think returning NULL ([to indicate failure](https://www.openssl.org/docs/man3.0/man3/CRYPTO_secure_malloc_init.html)) here might be the consistent way forward. We could also leave it as is and improve the documentation provided.